api/domain/ordersDomain.js

```diff
@@ -0,0 +1,168 @@
+    try {
+        return orderRepo.getAll()
+    } catch (error) {
+        throw new Error('No orders were found.')
+    }
+}
+
+exports.getById = async (orderId) => {
+    try {
+        await Verification.IdExists(Resources.Orders, orderId)
+
+        return orderRepo.getById(orderId)
+    } catch (error) {
+        throw new IdNotFound(error.message)
+    }
+}
+
+exports.addProducts = async (order, products) => {
+    try {
+        for (let n = 0; n < products.length; n++) {
+            const productsInOrder = await order.getProducts({ where: { id: products[n].id } })
+
+            if (productsInOrder.length !== 0) continue
+
+            await OrderHelpers.addQuantityAndTotalPrice(order, products[n].id, products[n].quantity)
+            await orderRepo.updateOrder(order.id, order.dataValues)
+            orderRepo.addProducts(order, products[n].id, products[n].quantity)
+        }
+    } catch (error) {
+        throw new IdNotFound(error.message)
+    }
+}
+
+exports.deleteProducts = async (orderId, products) => {
+    try {
+        const order = await orderRepo.getById(orderId)
+        
+        for (let n = 0; n < products.length; n++) {
+            const productsInOrder = await order.getProducts({ where: { id: products[n].id } })
+
+            if (productsInOrder.length === 0) continue
+
+            await OrderHelpers.substractQuantityAndTotalPrice(order, products[n].id)
+            await orderRepo.updateOrder(order.id, order.dataValues)
+            orderRepo.deleteProducts(order, products[n].id)
+        }
+    } catch (error) {
+        throw new IdNotFound(error.message)
+    }
+}
+
+exports.changeQuantity = async (order, products) => {
+    try {
+        for (let n = 0; n < products.length; n++) {
+            const productsInOrder = await order.getProducts({ where: { id: products[n].id } })
+
+            if (productsInOrder.length === 0) continue
+
+                await this.deleteProducts(order.id, products)
+                const updatedOrder = await orderRepo.getById(order.id)
+                this.addProducts(updatedOrder, products)
+        }
+    } catch (error) {
+        throw new IdNotFound(error.message)
+    }
+}
+
+exports.changeCurrency = async (prevOrder, updatedOrder) => {
+    for (let n = 0; n < updatedOrder.products.length; n++) {
+        await OrderHelpers.substractQuantityAndTotalPrice(prevOrder, updatedOrder.products[n].id)
+        
+        updatedOrder.dataValues.productsCount = prevOrder.dataValues.productsCount
+        updatedOrder.dataValues.productsTotalPrice = prevOrder.dataValues.productsTotalPrice
+        
+        await orderRepo.updateOrder(updatedOrder.id, updatedOrder.dataValues)
+
+        await orderRepo.deleteProducts(updatedOrder, updatedOrder.products[n].id)
+    }
+
+    for (let n = 0; n < updatedOrder.products.length; n++) {
+        await OrderHelpers.addQuantityAndTotalPrice(updatedOrder, updatedOrder.products[n].id, updatedOrder.products[n].products_orders.quantity)
```

Comment:
I'm not sure if you need to await methods from orderHelpers because they don't return any promise or anything, so maybe check if it works well without it and if so then remove awaits

Reply:
it is neccessary